consensus/enclave/impl/src/lib.rs

```diff
@@ -166,6 +168,25 @@ impl SgxConsensusEnclave {
         ))
     }
 
+    // Get a WellFormedTxContext for a Tx, given the minimum fee for its specified
+    // fee token.
+    fn get_well_formed_tx_context(&self, tx: &Tx, min_fee: u64) -> WellFormedTxContext {
+        // Compute the priority number to assign to this tx.
+        //
+        // Because different tokens have different minimum fees, and users usually
+        // use the minimum fee, we want to normalize so that the priority of any
+        // two payments using the minimum fee is exactly the same number.
+        //
+        // We also want to allow that increments of ~1% of the minimum fee actually
+        // affect the priority of your payment. So, we divide the min fee by 128,
+        // before dividing the fee by this. Separately, the fee map enforces that
+        // the minimum fees are >= SMALLEST_MINIMUM_FEE, and so we are not dividing by
+        // zero.
+        let (priority, _) = ct_u64_divide(tx.prefix.fee, min_fee >> SMALLEST_MINIMUM_FEE_LOG2);
```

Comment:
Couple of hypotheticals:
- First this is a really elegant simple design for ct fee priority design that captures sensible increases. Bravo.

First: Let's imagine a situation where mobilecoin or one of its sister tokens (meowblecoin or chowchow coin) takes off and becomes immensely popular and expensive. Bitcoin, ETH have had this problem and it's made the coin users cry.

1% of a minimum fee being $5 at some point and so it'd be helpful of minimum fee could be changed or the base <<7/128 modifier could. 

So questions:
1. Can minimum fee be adjusted at will? Who would be in charge of that? Would the consensus nodes need to agree on what the minimum fee is when they boot/reboot - or could this be something done whilst consensus nodes are live? 
2. Could the smallest_minimum_fee_log2 be adjusted per token? If there's a token with drastically different supply/tokenomics, 128 might not be negilible. 128 worked out nicely for MoB but perhaps we can make 128 scalable to the supply in the future?

Reply:
so I think the thing to keep in mind is, the range of a value is an entire u64, so for reference, this is like 4 billion squared.

I think it is fine to say that you lose ~2 decimal places out of 18 to allow that the minimum fee is >= 128. 16 decimal places of dynamic range is still insanely more dynamic range than any practical currency needs. People who make tokens and decide on how the scaling of smallest-representable-unit maps to user-facing unit, can just be sure to leave a little room at the end.

For instance, it was argued that for a dollar-based stablecoin, 6 decimals at the bottom is more than enough, perhaps overkill. That would make a minimum fee of 128u64 correspond to like, .00012 cents. And that leaves 12 decimal places at the top, so you could have a single TxOut worth a trillion dollars. Note that that is not imposing a market cap limit of 1 trillion dollars. It's just saying you can't represent more than 1 trillion dollars with a single TxOut.

Reply:
> Could the smallest_minimum_fee_log2 be adjusted per token?

I guess we could have a second CtTokenMap that maps from each token to this number.
If we really decide we need that flexibility I guess that's the approach I would take.
So then for some token where the 128 is no bueno, we could change it to like 8. Then you would have to increase fees in increments of 12% to see a priority change. Might be fine.

If you think this is a serious problem, we could make it 8 instead of 128 to start with. It's quite arbitrary. I don't honestly know how 1% increase increments vs. 12% increase increments would actually affect the characteristics of the fee market, I'm just guessing.

Reply:
More practically: a satoshi is `0.000_000_01` BTC, which is ~6 mils at 60kUSD/BTC (a "mil" is particularly evil unit in finance that means "basis points of a dollar/euro/gbp/etc.", aka `$0.0006`).

Our minimum fee in this scheme is `0.000_000_000_128`, which is 0.0128 satoshi, or 0.0768 mils (i.e. 7/100ths of 1/1000th of a dollar).